client/src/actions/actions.js

```diff
@@ -0,0 +1,91 @@
+import {
+    ADD_SONG,
+    BEGIN_DOWNLOAD_SONG_INFO,
+    DELETE_SONG, DOWNLOAD_SONG_INFO_ERROR, ERROR_CONFIRMED,
+    UPDATE_LINK,
+    UPDATE_SONG
+} from "./action-types";
+import config from "../config";
+import {formatString, getVideoIdFromLink} from "../services/formatting";
+
+export const updateLink = (newLink) => {
+    return {
+        type: UPDATE_LINK,
+        link: newLink
+    };
+};
+
+export const addSong = (link) => (dispatch) => {
+    dispatch({ type: BEGIN_DOWNLOAD_SONG_INFO, link });
+    const videoId = getVideoIdFromLink(link);
+    const url = formatString(config.server.resources.getVideoInfo, [videoId]);
+    fetch(url, {
+        method: 'GET',
+        headers: {
+            Accept: 'application/json'
+        },
+    })
+    .then(response => {
+        if (response.ok) {
+            response.json().then(songInfo => {
+                dispatch(downloadSongInfoSuccess(songInfo));
+            })
+        }
+        else {
+            response.json().then(jsonBody => {
+                dispatch(downloadSongInfoError(jsonBody.message));
+            })
+        }
+    })
+    .catch((e) => {
+        dispatch(downloadSongInfoError(config.download_song_info_error_message))
+    });
+};
+
+export const downloadSongInfoSuccess = (songInfo) => {
+    const song = {
+        id: songInfo['video_id'],
```

Comment:
You should mostly use camelCase in JS to prevent such cases.
Also, this would be a lot prettier:
```js
const song = { ...songInfo, chosenFormat: songInfo["formats"][0],  artist: songInfo["author"]['name'];
```
If you could just change the keys of the object `song` to match those of `songInfo`

Reply:
oh shit

Reply:
I can do it in one line but if you noticed I changed the almost every key of the json:
video_id => id
title => name
link => video_url
...

Reply:
Yes I see.
If you can *not* change it so it'll be cool - but it's not necessary. Just do it next time